python/py_vapid/__init__.py

```diff
@@ -52,41 +43,48 @@ def __init__(self, private_key=None):
 
         """
         self.private_key = private_key
+        if private_key:
+            self._public_key = self.private_key.public_key()
 
     @classmethod
-    def from_raw(cls, private_key):
+    def from_raw(cls, private_raw):
         """Initialize VAPID using a private key point in "raw" or
-        "uncompressed" form.
+        "uncompressed" form. Raw keys consist of a single, 32 octet
+        encoded integer.
 
-        :param private_key: A private key point in uncompressed form.
-        :type private_key: str
+        :param private_raw: A private key point in uncompressed form.
+        :type private_raw: bytes
 
         """
-        key = ecdsa.SigningKey.from_string(b64urldecode(private_key),
-                                           curve=cls._curve,
-                                           hashfunc=cls._hasher)
+        key = ec.derive_private_key(
+            int(binascii.hexlify(b64urldecode(private_raw)), 16),
+            curve=ec.SECP256R1(),
+            backend=default_backend())
         return cls(key)
 
     @classmethod
     def from_pem(cls, private_key):
         """Initialize VAPID using a private key in PEM format.
 
         :param private_key: A private key in PEM format.
-        :type private_key: str
+        :type private_key: bytes
 
         """
-        key = ecdsa.SigningKey.from_pem(private_key)
-        return cls(key)
+        # not sure why, but load_pem_private_key fails to deserialize
```

Comment:
It's in the test: https://github.com/web-push-libs/vapid/pull/42/files#diff-f4037d7f16c50099d22585d0d3b77b90R30

I'm constructing the PEM from the b64 encoded DER, which is pretty much what a PEM is. That's why I can strip the header/footer and then concat the strings to decode into a DER.